controllers/logicalvolume_controller.go

```diff
@@ -60,9 +61,9 @@ func (r *LogicalVolumeReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	}
 
 	if lv.ObjectMeta.DeletionTimestamp == nil {
-		if !containsString(lv.Finalizers, topolvm.GetLogicalVolumeFinalizer()) {
+		if !controllerutil.ContainsFinalizer(lv, topolvm.GetLogicalVolumeFinalizer()) {
```

Comment:
`AddFinalizer` function checks the duplication of finalizers. And You can know from the return value if the finalizer already existed or not. So you can write it more simpler.
https://github.com/kubernetes-sigs/controller-runtime/blob/c304e7ec2ee7879f53ce69c12a7aa133853fccdb/pkg/controller/controllerutil/controllerutil.go#L353

Reply:
I considered the option but did not choose it due to performance. 
`AddFinaizer` or `RemoveFinalizer` overwrites its argument, so we need to `DeepCopy` to create a patch before calling it. `DeepCopy` traverses whole object fields, but `AddFinaizer` or `RemoveFinalizer` does only finalizers, the cost is lower than `DeepCopy`.